packages/arb-ts/src/lib/inbox/inbox.ts

```diff
@@ -0,0 +1,226 @@
+  ): Promise<ForceInclusionParams | null> {
+    const bridge = Bridge__factory.connect(
+      this.l2Network.ethBridge.bridge,
+      this.l1Provider
+    )
+
+    const eFetcher = new EventFetcher(this.l1Provider)
+
+    // events dont become eligible until they pass a delay
+    // find a block range which will emit eligible events
+    const blockRange = await this.getEligibleBlockRange(searchRangeBlocks)
+
+    // get all the events in this range
+    const events = await eFetcher.getEvents<Bridge, MessageDeliveredEvent>(
+      bridge.address,
+      Bridge__factory,
+      b => b.filters.MessageDelivered(),
+      { fromBlock: blockRange.startBlock, toBlock: blockRange.endBlock }
+    )
+
+    // no events appeared within that time period
+    if (events.length === 0) return null
+
+    // take the last event - as including this one will include all previous events
+    const eventInfo = events[events.length - 1]
+    const sequencerInbox = SequencerInbox__factory.connect(
+      this.l2Network.ethBridge.sequencerInbox,
+      this.l1Provider
+    )
+    // has the sequencer inbox already read this latest message
+    const totalDelayedRead = await sequencerInbox.totalDelayedMessagesRead()
+    if (totalDelayedRead.gt(eventInfo.event.messageIndex)) {
+      // nothing to read - more delayed messages have been read than this current index
+      return null
+    }
+
+    const delayedAcc = await bridge.inboxAccs(eventInfo.event.messageIndex)
+
+    return { ...eventInfo, delayedAcc: delayedAcc }
+  }
+
+  /**
+   * Force includes all eligible messages in the delayed inbox.
+   * The inbox contract doesnt allow a message to be force-included
+   * until after a delay period has been completed.
+   * @param messageDeliveredEvent Provide this to include all messages up to this one. Responsibility is on the caller to check the eligibility of this event.
+   * @returns The force include transaction, or null if no eligible message were found for inclusion
+   */
+  public async forceInclude<T extends ForceInclusionParams | undefined>(
+    messageDeliveredEvent?: T,
+    overrides?: Overrides
+  ): Promise<
+    // if a message delivered event was supplied then we'll definately return
+    // a contract transaction or throw an error. If one isnt supplied then we may
+    // find no eligible events, and so return null
+    T extends ForceInclusionParams
+      ? ContractTransaction
+      : ContractTransaction | null
+  >
+  public async forceInclude<T extends ForceInclusionParams | undefined>(
+    messageDeliveredEvent?: T,
+    overrides?: Overrides
+  ): Promise<ContractTransaction | null> {
+    const sequencerInbox = SequencerInbox__factory.connect(
+      this.l2Network.ethBridge.sequencerInbox,
+      this.l1Signer
+    )
+    const eventInfo =
+      messageDeliveredEvent || (await this.getForceIncludeableEvent())
+
+    if (!eventInfo) return null
+    const block = await this.l1Provider.getBlock(eventInfo.blockHash)
```

Comment:
we can get the blockhash as part of a multicall in the `getForceIncludeableEvent` in order to save a network query here

Reply:
I think you meant timestamp here - updated.

Reply:
oops, ye